src/compiler/cpp_generator.cc

```diff
@@ -993,7 +997,15 @@ void PrintHeaderServerMethodCallback(
         "controller) {\n"
         "               return this->$"
         "Method$(context, request, response, controller);\n"
-        "             }));\n");
+        "             }));\n}\n");
+    printer->Print(*vars,
+                   "void SetMessageAllocatorFor_$Method$(\n"
+                   "    ::grpc::experimental::MessageAllocator< "
+                   "$RealRequest$, $RealResponse$>* allocator) {\n"
+                   "  dynamic_cast<::grpc::internal::CallbackUnaryHandler< "
```

Comment:
Err, do we need a dynamic_cast here? We haven't used them anywhere yet and I'm not a fan.

Reply:
Actually, I feel that static_cast should work perfectly since this is a downward cast in the same hierarchy.